build.rs

```diff
@@ -0,0 +1,44 @@
+use std::env;
+use std::process::Command;
+use std::str::{self, FromStr};
+
+fn main(){
+    let minor = match rustc_minor_version() {
+        Some(minor) => minor,
+        None => return,
+    };
+
+    // const fn stabilized in Rust 1.31:
+    if minor >= 31 {
+        println!("cargo:rustc-cfg=const_fn");
```

Comment:
Following the [example of `regex`](https://github.com/rust-lang/regex/blob/master/build.rs#L16-L29) I think we should call this `bitflags_const_fn` so it's namespaced just for `bitflags` and clear that it's a cfg we're setting.